sql/range_test.go

```diff
@@ -767,3 +811,325 @@ func and(expressions ...sql.Expression) sql.Expression {
+				"",
+		},
+		{
+			name:      "insert largest",
+			setupRngs: []sql.Range{r(req(7)), r(req(3)), r(req(11))},
+			setupExp: "RangeColumnExprTree\n" +
+				"│   ┌── [11, 11] max: >11 color: 1\n" +
+				"└── [7, 7] max: >11 color: 0\n" +
+				"    └── [3, 3] max: >3 color: 1\n" +
+				"",
+			rng: r(req(14)),
+			exp: "RangeColumnExprTree\n" +
+				"│       ┌── [14, 14] max: >14 color: 1\n" +
+				"│   ┌── [11, 11] max: >14 color: 0\n" +
+				"└── [7, 7] max: >14 color: 0\n" +
+				"    └── [3, 3] max: >3 color: 0\n" +
+				"",
+		},
+		{
+			name:      "insert rebalance left child",
+			setupRngs: []sql.Range{r(req(7)), r(req(3)), r(req(11)), r(req(0))},
+			setupExp: "RangeColumnExprTree\n" +
+				"│   ┌── [11, 11] max: >11 color: 0\n" +
+				"└── [7, 7] max: >11 color: 0\n" +
+				"    └── [3, 3] max: >3 color: 0\n" +
+				"        └── [0, 0] max: >0 color: 1\n" +
+				"",
+			rng: r(req(1)),
+			exp: "RangeColumnExprTree\n" +
+				"│   ┌── [11, 11] max: >11 color: 0\n" +
+				"└── [7, 7] max: >11 color: 0\n" +
+				"    │   ┌── [3, 3] max: >3 color: 1\n" +
+				"    └── [1, 1] max: >3 color: 0\n" +
+				"        └── [0, 0] max: >0 color: 1\n" +
+				"",
+		},
+		{
+			name:      "insert rebalance right child",
+			setupRngs: []sql.Range{r(req(7)), r(req(3)), r(req(11)), r(req(12))},
+			setupExp: "RangeColumnExprTree\n" +
+				"│       ┌── [12, 12] max: >12 color: 1\n" +
+				"│   ┌── [11, 11] max: >12 color: 0\n" +
+				"└── [7, 7] max: >12 color: 0\n" +
+				"    └── [3, 3] max: >3 color: 0\n" +
+				"",
+			rng: r(req(13)),
+			exp: "RangeColumnExprTree\n" +
+				"│       ┌── [13, 13] max: >13 color: 1\n" +
+				"│   ┌── [12, 12] max: >13 color: 0\n" +
+				"│   │   └── [11, 11] max: >11 color: 1\n" +
+				"└── [7, 7] max: >13 color: 0\n" +
+				"    └── [3, 3] max: >3 color: 0\n" +
+				"",
+		},
+		{
+			name:      "insert rebalance root from left",
+			setupRngs: []sql.Range{r(req(7)), r(req(3)), r(req(11)), r(req(0)), r(req(1)), r(req(2)), r(req(4))},
+			setupExp: "RangeColumnExprTree\n" +
+				"│   ┌── [11, 11] max: >11 color: 0\n" +
+				"└── [7, 7] max: >11 color: 0\n" +
+				"    │       ┌── [4, 4] max: >4 color: 1" +
+				"\n    │   ┌── [3, 3] max: >4 color: 0\n" +
+				"    │   │   └── [2, 2] max: >2 color: 1\n" +
+				"    └── [1, 1] max: >4 color: 1\n" +
+				"        └── [0, 0] max: >0 color: 0\n" +
+				"",
+			rng: r(req(5)),
+			exp: "RangeColumnExprTree\n" +
+				"│       ┌── [11, 11] max: >11 color: 0\n" +
+				"│   ┌── [7, 7] max: >11 color: 1\n" +
+				"│   │   │   ┌── [5, 5] max: >5 color: 1\n" +
+				"│   │   └── [4, 4] max: >5 color: 0\n" +
+				"└── [3, 3] max: >11 color: 0\n" +
+				"    │   ┌── [2, 2] max: >2 color: 0\n" +
+				"    └── [1, 1] max: >2 color: 1\n" +
+				"        └── [0, 0] max: >0 color: 0\n" +
+				"",
+		},
+		{
+			name:      "insert rebalance root from right",
+			setupRngs: []sql.Range{r(req(7)), r(req(3)), r(req(11)), r(req(8)), r(req(9)), r(req(10)), r(req(12))},
```

Comment:
I'd like some tests that use more than equals. We can have tests of all less than, etc. and also combine them. Hypothetically, it could all work great for equals since it's the most basic case, and break down on the other types.